src/RESTController.js

```diff
@@ -110,11 +110,14 @@ const RESTController = {
           let response;
           try {
             response = JSON.parse(xhr.responseText);
+            const availableHeaders = xhr.getAllResponseHeaders() || "";
             headers = {};
-            if (typeof xhr.getResponseHeader === 'function' && xhr.getResponseHeader('access-control-expose-headers')) {
+            if (typeof xhr.getResponseHeader === 'function' && availableHeaders.indexOf('access-control-expose-headers') >= 0) {
```

Comment:
I think you should add a check for `typeof xhr. getAllResponseHeaders === 'function'`. This reduces the number of changes to the test suite and also check for xhr libraries that don't support getAllResponseHeaders.

Reply:
would you just ignore headers completely then if `xhr.getAllResponseHeaders !== 'function'` ?

Reply:
```
const availableHeaders = typeof xhr.getAllResponseHeaders === 'function' ? xhr.getAllResponseHeaders() : "";
```
?

Reply:
Yes thats better. Now you can revert the tests and only mock `getAllResponseHeaders` where you need it.